src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using Xunit;
+// ReSharper disable InconsistentNaming
+
+namespace Tests.Collections
+{
+    public enum CollectionOrder
+    {
+        Unspecified,
+        Sequential
+    }
+
+    public static class CollectionAssert
+    {
+        public static void Equal(IEnumerable expected, IEnumerable actual)
+        {
+            Assert.Equal(expected.OfType<object>(), actual.OfType<object>());
+        }
+
+        public static void Equal(IList expected, IList actual)
+        {
+            Equal((IEnumerable)expected, actual);
+            Assert.Equal(expected.Count, actual.Count);
+            for (int i = 0; i < expected.Count; i++)
+            {
+                var expectedArray = expected as Array;
+                var actualArray = expected as Array;
+                var expectedLowerBound = expectedArray == null
+                                             ? 0
+                                             : expectedArray
+                                                   .GetLowerBound(0);
+                var actualLowerBound = actualArray == null
+                                           ? 0 : actualArray.GetLowerBound(
+                                               0);
+                Assert.Equal(expected[i + expectedLowerBound], actual[i + actualLowerBound]);
+            }
+        }
+    }
+
+    public static class IListExtensions
+    {
+        public static void AddRange(this IList list, IEnumerable items)
+        {
+            foreach (var item in items)
+                list.Add(item);
+        }
+
+        public static void InsertRange(this IList list, object[] items)
```

Comment:
Looking at the code, I think this is "prepend"?  IList doesn't have an implied position for insertion, so this doesn't really have a pigeon hole in which to fit.